docs/source/workflow/generate-types.mdx

```diff
@@ -0,0 +1,205 @@
+
+### Installing and configuring dependencies
+
+Run the following command to install the `@graphql-codegen/cli`, `@graphql-codegen/typescript`, and `@graphql-codegen/typescript-resolvers` packages into your project's dev dependencies:
+
+```bash
+npm install -D @graphql-codegen/cli @graphql-codegen/typescript @graphql-codegen/typescript-resolvers
+```
+
+> For more information on each package above, check out the [GraphQL Code Generator](https://www.the-guild.dev/graphql/codegen/plugins) docs.
+
+Next, we'll set up a configuration file to tell GraphQL Code Generator where and how to generate types. You can do this by manually creating a `codegen.yml` file or by using the following command, which walks you through the process:
+
+```bash
+npx graphql-code-generator init
+```
+
+Below is an example of a `codegen.yml` file:
+
+```yaml
+# This configuration file tells GraphQL Code Generator how
+# to generate types based on our schema.
+schema: "./src/schema.ts"
+generates:
+  # Specify where our generated types should live.
+  ./src/__generated__/resolvers-types.ts:
+    plugins:
+      - "typescript"
+      - "typescript-resolvers"
+    config:
+      useIndexSignature: true
+      # More on this below!
+      contextType: ../index#MyContext
+```
+
+> [See the docs](https://www.the-guild.dev/graphql/codegen/plugins/typescript-resolvers#usage-example) for more information on the above configuration options.
+
+Finally, we recommend adding helpful scripts to your `package.json` file to ensure your TS types are regularly generated:
+
+```json
+{
+// ...
+  "scripts": {
+    "generate": "graphql-codegen --config codegen.yml",
+    "compile": "npm run generate && tsc",
+    "postinstall": "npm run compile",
+    "start": "npm run compile && node ./dist/index.js",
+  },
+// ...
+}
+```
+
+Above, running the `npm start` command generates types based on our GraphQL schema _and_ compiles our TypeScript code. The first time you run the `graphql-codegen` command, you'll see a file full of generated types at the path you specified in your `codegen.yml` file. 
+
+## Adding types to resolvers
+
+The `typescript-resolvers` plugin creates a `Resolvers` type that you can use to add a type to your [resolver map](../data/resolvers#base-syntax), ensuring your resolvers return values match the field types specified by your schema.
+
+Import the `Resolvers` type into the file where you define your resolvers:
+
+```ts title="resolvers.ts"
+// This is the file where our generated types live
+// (specified in our `codegen.yml` file)
+import { Resolvers } from './__generated__/resolvers-types';
+```
+
+You can now add the `Resolvers` type directly to your resolver map:
+
+```ts
+export const resolvers: Resolvers = {}
+```
+
+Your resolvers can now type check that the *arguments* and *return value* for each resolver match the schema:
```

Comment:
I think it would probably be helpful to show how to split resolvers across files (eg one object type per file).